### PR TITLE
Linux: add AppStream metadata

### PR DESCRIPTION
**Pull request by eszlari**
Also rename `UI/dist/obs.desktop` according to the spec:

https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html

### CODE REVIEW

File: UI/xdg-data/com.obsproject.Studio.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  </description>
+  <url type="homepage">https://obsproject.com</url>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://obsproject.com/assets/images/OBSDemoApp.jpg</image>
+    </screenshot>
+  </screenshots>
+  <content_rating type="oars-1.1"/>
+  <releases>
+    <release version="23.0.2" date="2019-03-08"/>
```
Comment (WizardCM) [inline UI/xdg-data/com.obsproject.Studio.appdata.xml]: This should be auto-generated.

Comment (eszlari) [reply UI/xdg-data/com.obsproject.Studio.appdata.xml]: But please keep the build reproducible!

Comment (WizardCM) [reply UI/xdg-data/com.obsproject.Studio.appdata.xml]: I mean something like this https://github.com/obsproject/obs-studio/pull/1629/files#diff-af3b638bc2a3e6c650974192a53c7291R42

It does a string replace to generate a final file.

Comment (eszlari) [reply UI/xdg-data/com.obsproject.Studio.appdata.xml]: I know. You could replace the version string, but the date can't be a build timestamp.

Multiple release entries are also possible:
```
<releases>
  <release version="23.0.2" date="2019-03-08"/>
  <release version="23.0.1" date="2019-XX-XX"/>
  <release version="23.0.0" date="2019-XX-XX"/>
  ...
</releases>
```

File: UI/xdg-data/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+install(FILES com.obsproject.Studio.desktop DESTINATION ${CMAKE_INSTALL_FULL_DATAROOTDIR}/applications)
+install(FILES com.obsproject.Studio.appdata.xml DESTINATION ${CMAKE_INSTALL_FULL_DATAROOTDIR}/metainfo)
+install(FILES ../forms/images/obs.png DESTINATION ${CMAKE_INSTALL_FULL_DATAROOTDIR}/icons/hicolor/256x256/apps)
+
```
Comment (RytoEX) [inline UI/xdg-data/CMakeLists.txt]: Style:  While files must end in a newline character, they don't need an extra blank newline.  You can remove this line.

```suggestion
```

File: UI/xdg-data/com.obsproject.Studio.appdata.xml (hunk 1 above)
Comment (RytoEX) [reply UI/xdg-data/com.obsproject.Studio.appdata.xml]: You can use CMake to generate the current date of the CMake run.
```
string(TIMESTAMP TODAY "%Y-%m-%d")
```
Generating multiple release elements could be done by either:

1. We maintain a public list of release dates/elements to be fetched on build
2. The build process iterates through all git tags and uses the tag date as the "release date"

Comment (eszlari) [reply UI/xdg-data/com.obsproject.Studio.appdata.xml]: Using the current date would make the build non-reproducible.

Comment (RytoEX) [reply UI/xdg-data/com.obsproject.Studio.appdata.xml]: You could still use the git tag date, which would be immutable.

Comment (WizardCM) [reply UI/xdg-data/com.obsproject.Studio.appdata.xml]: That's what I'd do - the git tag with its date, unless the user is building a non-tagged commit, then build with today's date.

**Comment by eszlari** [comment]
@WizardCM @RytoEX I made the `appdata.xml` file auto-generated.

**Pull request merged**